packages/client/src/components/app/Modal.svelte

```diff
@@ -0,0 +1,162 @@
+<script>
+  import { fade, fly } from "svelte/transition"
+  import { getContext } from "svelte"
+  import { clickOutside, Icon } from "@budibase/bbui"
+  import Portal from "svelte-portal"
+
+  const component = getContext("component")
+  const { styleable, modalStore, builderStore, dndIsDragging } =
+    getContext("sdk")
+
+  export let onClose
+  export let ignoreClicksOutside
+  export let size
+
+  // Open modal automatically in builder
+  $: {
+    if ($builderStore.inBuilder) {
+      if (
+        $component.inSelectedPath &&
+        $modalStore.contentId !== $component.id
+      ) {
+        modalStore.actions.open($component.id)
+      } else if (
+        !$component.inSelectedPath &&
+        $modalStore.contentId === $component.id &&
+        !$dndIsDragging
+      ) {
+        modalStore.actions.close()
+      }
+    }
+  }
+
+  $: open = $modalStore.contentId === $component.id
+
+  const handleModalClose = async () => {
+    if (onClose) {
+      await onClose()
+    }
+    modalStore.actions.close()
+  }
+</script>
+
+{#if open}
+  <Portal target=".modal-portal">
+    <div
+      transition:fade={{ duration: 300 }}
+      class="modal-scroll-container"
+      class:open={$modalStore.open}
+    >
+      <div class="modal-container">
+        <div
+          transition:fly={{ duration: 300, y: 300 }}
+          use:styleable={$component.styles}
+          use:clickOutside={!ignoreClicksOutside ? handleModalClose : null}
+          class="modal {size}"
+        >
+          <div class="modal-header">
+            <Icon
+              color="var(--spectrum-global-color-gray-600)"
+              name="Close"
+              hoverable
+              on:click={handleModalClose}
+            />
+          </div>
+          <div class="modal-main">
+            <div class="modal-main-inner">
+              <slot />
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </Portal>
+{/if}
+
+<style>
+  .modal-scroll-container {
+    background-color: #00000078;
```

Comment:
Agreed, we should use the same styles as our BBUI modal as much as possible.

Reply:
I was just eyeballing it based on the mockups in the ticket, if anything I'd say maybe the current overlay is a little lighter compared to the mockup.

Reply:
Consistency is the most important thing, so lets just use everything we currently use in the existing modal (otherwise users will see 2 different styles of modals depending on whether they're using this component or seeing a default modal for a confirmation prompt etc).